client/header/activity-panel/tabs/index.js

```diff
@@ -0,0 +1,67 @@
+import { NavigableMenu } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+import { Tab } from '../tab';
+import { recordEvent } from 'lib/tracks';
+
+export const Tabs = ( {
+	tabs,
+	onTabClick,
+	selectedTab: selectedTabName,
+	tabOpen = false,
+} ) => {
+	const [ tabState, setTabState ] = useState( {
+		tabOpen,
+		currentTab: selectedTabName,
+	} );
+
+	// Keep state synced with props
+	useEffect( () => {
+		setTabState( {
+			tabOpen,
+			currentTab: selectedTabName,
+		} );
+	}, [ tabOpen, selectedTabName ] );
+
+	return (
+		<NavigableMenu
+			role="tablist"
+			orientation="horizontal"
+			className="woocommerce-layout__activity-panel-tabs"
+		>
+			{ tabs &&
```

Comment:
With this refactor, we could now have a filter right here for the tabs. Given the uncertain future of the activity panels though, my vote is to not add it quite yet - but still really cool that your changes have allowed for a good spot to add it in.